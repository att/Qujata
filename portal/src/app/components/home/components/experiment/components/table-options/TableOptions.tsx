import styles from './TableOptions.module.scss';
import cn from 'classnames';
import { TABLE_OPTIONS_EN } from './translate/en';
import SelectColumnsSvg from '../../../../../../../assets/images/select-columns.svg';
import SelectColumnsCheckedSvg from '../../../../../../../assets/images/select-columns-checked.svg';
import { Button, ButtonActionType, ButtonSize, ButtonStyleType } from '../../../../../../shared/components/att-button';

const SelectColumnsAriaLabel: string = TABLE_OPTIONS_EN.SELECT_COLUMNS;

export interface TableOptionsProps {
  handleSelectColumnsClick: () => void;
  isPopupOpen: boolean;
}

export const TableOptions: React.FC<TableOptionsProps> = (props: TableOptionsProps) => {
  return (
    <div className={styles.table_options_wrapper}>
      <Button
        ariaLabel={SelectColumnsAriaLabel}
        size={ButtonSize.NONE}
        styleType={ButtonStyleType.WRAPPER}
        actionType={ButtonActionType.BUTTON}
        onButtonClick={props.handleSelectColumnsClick}
      >
        <div className={styles.option_wrapper}>
          <img className ={cn(styles.option_image, styles.default_image)} src={SelectColumnsSvg} alt={SelectColumnsAriaLabel} />
          <img className ={cn(styles.option_image, styles.hover_image)} src={SelectColumnsCheckedSvg} alt={SelectColumnsAriaLabel} />
          <span className={styles.option_text}>{TABLE_OPTIONS_EN.SELECT_COLUMNS}</span>
        </div>
      </Button>
    </div>
  );
};
