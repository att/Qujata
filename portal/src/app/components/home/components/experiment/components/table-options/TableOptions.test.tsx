import { render } from '@testing-library/react';
import { TableOptions } from './TableOptions';

jest.mock('../hooks/useExperimentData');

describe('TableOptions', () => {
    test('should pass', async () => {
      expect(true).toBe(true);
    });
});
