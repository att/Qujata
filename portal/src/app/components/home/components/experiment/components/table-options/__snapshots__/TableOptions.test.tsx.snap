// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableOptions renders without crashing 1`] = `
<div
  class="table_options_wrapper"
>
  <a
    class="link_wrapper text_link"
    href="http://localhost:3000/d/b3adde71-2231-4c2f-b216-3af0a16d8d9v/qujata-analysis?orgId=1&from=2021-07-26T12:00:00.000Z&to=2021-07-26T12:00:00.000Z"
    rel="noopener"
    target="_blank"
  >
    <div
      class="grafana_link"
    >
      <img
        alt="eye"
        class="eye_icon"
        src="eye.svg"
      />
      View in Grafana
    </div>
  </a>
  <button
    aria-label="Select columns"
    class="button wrapper_button"
    type="button"
  >
    <div
      class="options_wrapper"
    >
      <img
        alt="Select columns"
        class="select_columns_icon"
        src="select-columns.svg"
      />
      <span
        class="select_columns_text"
      >
        Select columns
      </span>
    </div>
  </button>
</div>
`;
